space_allocator.py

```diff
@@ -0,0 +1,171 @@
+"""
+Welcome to the Space allocator.
+Usage:
+    space_allocator create_room <room_type> <room_name>...
+    space_allocator add_person <first_name> <last_name> <person_type> [<wants_accommodation>]
+    space_allocator print_room <room_name>
+    space_allocator print_allocations [<file_name>] [--table]
+    space_allocator print_unallocated [<file_name>]
+    space_allocator reallocate_person <person_identifier> <new_room_name>
+    space_allocator load_people <file_name>
+    space_allocator save_state [<sqlite_database>]
+    space_allocator load_state [<sqlite_database>]
+    space_allocator (-i | --interactive)
+    space_allocator (-h | --help | --version)
+Options:
+    --version  show program's version number and exit
+    --table  Prints out a table on the screen.
+    -i, --interactive  Interactive Mode
+    -h, --help  Show this screen and exit.
+"""
+
+import sys
+import cmd
+from docopt import docopt, DocoptExit
+import colorful
+from src.dojo import Dojo
+
+dojo = Dojo()
+
+
+def docopt_cmd(func):
+    """
+    This decorator is used to simplify the try/except block and pass the result
+    of the docopt parsing to the called action.
+    """
+
+    def fn(self, arg):
```

Comment:
It's best practice to clearly name your functions with descriptive names not initials